File: Whiteboard practice app/DS&As/Trees/Problems/MinDepthOfBinaryTree.cs
public class BinaryTree
{
    public int GetMinDepth(TreeNode root)
    {
        if(root == null) return 0;

        Queue<TreeNode> nodeQ = new();
        nodeQ.Enqueue(root);
        int depth = 1;

        while(nodeQ.Count > 0)
        {
            int levelSize = nodeQ.Count;
            while(levelSize-- > 0)
            {
                TreeNode curr = nodeQ.Dequeue();
                if(curr.left == null && curr.right == null) return depth;
                if(curr.left != null) nodeQ.Enqueue(curr.left);
                if(curr.right != null) nodeQ.Enqueue(curr.right);
            }
            depth++;
        }

        return depth;
    }
}

File: Whiteboard practice app/DS&As/Trees/Problems/MinDepthOfBinaryTree.js
var minDepth = function(root) {
    if(!root) return 0;
    
    let queue = [root];
    let depth = 1;
    let curr;
    while(queue.length > 0)
    {
        let levelSize = queue.length;
        while(levelSize > 0){
            curr = queue.shift();
            if(!curr.left && !curr.right) return depth;
            if(curr.left) queue.push(curr.left);
            if(curr.right) queue.push(curr.right);
            levelSize--;
        }
        depth++;
    }
    
    return depth;
};

/* 
implement bfs, return the level where the first leaf node appears

    to implement bfs, we need to have a queue
    while iterate over the queue
    set current level length = queue size
    while level length > 0, we iterate over the level
    we will check if curr has a left or rigth node
    if not, then return depth
    if one or both do, we push the left &/or right node 
    subtract 1 from level length for each iteration
*/